### PR TITLE
 JAMES-3369 Email/get preview & hasAttachment simple implementation

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
Deployment tests

test this please

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala [hunk 1]
```diff
@@ -2622,6 +2622,232 @@ trait EmailGetMethodContract {
...
+    assertThatJson(response)
+      .inPath("methodResponses[0][1].list[0]")
+      .isEqualTo(s"""{
+         |    "id": "1",
+         |    "hasAttachment": false
+         |}""".stripMargin)
+  }
+
+  @Test
+  def hasAttachmentForMultipart(server: GuiceJamesServer): Unit = {
```
Comment (remk) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala:2809]: maybe add a test for a multipart mail without attachements

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala [hunk 2]
```diff
@@ -77,7 +79,9 @@ case class EmailGetRequest(accountId: AccountId,
         htmlBody = bodyStructure.htmlBody,
         attachments = bodyStructure.attachments,
         headers = asEmailHeaders(mime4JMessage.getHeader),
-        bodyValues = bodyValues)
+        bodyValues = bodyValues,
+        hasAttachment = HasAttachment(!firstMessage.getLoadedAttachments.isEmpty),
```
Comment (remk) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala:83]: you can use nonEmpty if it is a scala collection

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala:83]: It is not a scala collection. .asScala.nonEmpty seems overkill...

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala [hunk 3]
```diff
@@ -2622,6 +2622,232 @@ trait EmailGetMethodContract {
...
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1].list[0]")
+      .isEqualTo(
+        s"""{
+           |     "id": "1",
+           |     "preview": "A HTML body..."
+           |}""".stripMargin)
+  }
+
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala:2762]: Should we test that the preview gets truncated if more than 256 characters, as specified in the specs?

**Review by Arsnael — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**